openapi.yaml

```diff
@@ -0,0 +1,484 @@
+                $ref: '#/components/schemas/ClusterId'
+              example:
+                id: cee0oh
+        '400':
+          description: Cluster cannot be created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Reason'
+        '405':
+          description: Cluster already exists
+  /clusters/{clusterId}:
+    get:
+      summary: Returns all values of a cluster
+      tags:
+        - cluster 
+      parameters:
+        - $ref: '#/components/parameters/ClusterId'
+      responses:
+        '200':
+          description: Cluster found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Cluster'
+        '404':
+          description: A cluster with the specified id was not found.
+    patch:
+      summary: Updates a cluster
+      tags:
+        - cluster
+      parameters:
+        - $ref: '#/components/parameters/ClusterId'
+      requestBody:
+        description: Update cluster with properties to be changed (RFC 7396)
+        required: true
+        content:
+          application/merge-patch+json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/Cluster'
+                - type: object
+                  properties:
+                    displayName:
+                      nullable: true
+      responses:
+        '204':
+          description: Cluster updated
+        '403':
+          description: Cluster update forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Reason'
+    delete:
+      summary: Deletes a cluster
+      tags:
+        - cluster
+      responses:
+        '204':
+          description: Cluster deleted
+        '403':
+          description: Cluster deletion forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Reason'
+  /install/syn.yaml:
+    get:
+      summary: Returns the Steward installation YAML
+      description: |
+        Autogenerated YAML containing all the needed parameters for having Steward up and running
+        * Iterates through all available Cluster objects to find the object matching the token in the field `spec.bootstrapToken.token`
+        * Checks if the token is valid (fields `spec.bootstrapToken.valid` and field `spec.bootstrapToken.validUntil`)
+        * If valid, generates the installation YAML based on a template to install Steward in the cluster
+        * If valid and YAML could be successfully delivered, set the field `spec.bootstrapToken.valid` to `false`
+      tags:
+        - bootstrapping
+      parameters:
+        - in: query
+          name: token
+          schema:
+            type: string
+          description: Initial bootstrap token
+      responses:
+        '200':
+          description: Cluster found and token valid
```

Comment:
Should this specify content type YAML?

Reply:
With this question, you have opened the box of pandora :see_no_evil: https://stackoverflow.com/questions/332129/yaml-mime-type -> Looks like there is no simple answer to that. I would propose that we do not return a YAML here but a JSON file, that's good for K8s as well and would simply not be an issue with the media type anymore. What do you think?

Reply:
haha, good idea. Let's go with JSON :+1: